slang-api/src/test/java/com/sonarsource/slang/impl/ReturnTreeImplTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.impl;
+
+import com.sonarsource.slang.api.LiteralTree;
+import com.sonarsource.slang.api.Token;
+import com.sonarsource.slang.api.TreeMetaData;
+import org.junit.Test;
+
+import static com.sonarsource.slang.utils.SyntacticEquivalence.areEquivalent;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ReturnTreeImplTest {
+  @Test
+  public void test() {
+    TreeMetaData meta = null;
+    TokenImpl returnKeyword = new TokenImpl(new TextRangeImpl(1, 0, 1, 3), "return", Token.Type.KEYWORD);
+    ReturnTreeImpl returnTree = new ReturnTreeImpl(meta, returnKeyword, null);
+
+    assertThat(returnTree.children()).hasSize(0);
+    assertThat(returnTree.keyword().text()).isEqualTo("return");
+    assertThat(returnTree.body()).isNull();
+
+    ReturnTreeImpl returnTree2 = new ReturnTreeImpl(meta, returnKeyword, new LiteralTreeImpl(meta, "foo"));
```

Comment:
you can reuse previous var and avoid this `2` suffix. Or better give meaningful name in both cases, e.g. `returnWithoutValue` and `returnWithValue`